include/onnxruntime/core/common/spin_pause.h

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#if defined(__x86_64__)
```

Comment:
This should be the below to make it work with MSVC too:

#if defined(_M_AMD64) || defined(__x86_64__)

Reply:
Thanks; I has a related question I was going to check on before submitting -- do you know an equivalent to mm_pause we should use on ARM?

Reply:
ARM does have a yield instruction that is a nop hint to the processor that the current hardware thread can be swapped out. MSVC has a __yield intrinisic to use this. With gcc/clang, you can do __asm__ __volatile__ ("yield");. I don't have an ARM64 system that has multiple hardware threads to measure the impact of this. Maybe just leave it out for now?